feat(ml-dsa): Implement verifying_key method for SigningKey

romen opened the pull request:
Introduce a utility method to derive a `VerifyingKey` from a `SigningKey`, even when the original seed or precomputed public key is unavailable. This aids in scenarios where private keys are imported from external sources lacking associated public key data.

tarcieri commented:
It would be good to add an impl of the [`signature::Keypair`] trait: https://docs.rs/signature/3.0.0-rc.1/signature/trait.Keypair.html

You can keep the inherent method too so it still works with the trait out-of-scope

romen commented:
@tarcieri sorry, it took some time.

I rebased this PR on latest master, and added the trait implementation.

The inherent method is now just a convenience shim around calling the trait implementation.
